src/trio/testing/_raises_group.py

```diff
@@ -254,13 +267,15 @@ def __str__(self) -> str:
 # We lie to type checkers that we inherit, so excinfo.value and sub-exceptiongroups can be treated as ExceptionGroups
 if TYPE_CHECKING:
     SuperClass = BaseExceptionGroup
-# Inheriting at runtime leads to a series of TypeErrors, so we do not want to do that.
 else:
-    SuperClass = Generic
+    # At runtime, just discard this base class.
+    SuperClass = EmptySuperclass()
 
 
 @final
-class RaisesGroup(ContextManager[ExceptionInfo[BaseExceptionGroup[E]]], SuperClass[E]):
+class RaisesGroup(
+    SuperClass[E], ContextManager[ExceptionInfo[BaseExceptionGroup[E]]], Generic[E]
+):
```

Comment:
Why do we need `Generic` here?

Reply:
I was thinking that generic classes need the `Generic` subclassing, but they don't. So some of this can be removed.

Reply:
I might have added `Generic` as a superclass precisely because I wanted a superclass that did nothing (because its other superclasses themselves already inherit from it)